pkg/client/add_events.go

```diff
@@ -40,15 +40,21 @@ import (
 Wrapper around: https://app.scalyr.com/help/api#addEvents
 */
 
-// AddEvents enqueues all the events for processing.
-// It returns an error if the batch was not accepted.
-// When you want to finish processing, call Finish method.
+// AddEvents enqueues given events for processing (sending to Dataset).
+// It returns an error if the batch was not accepted (eg: client is being shutdown).
+// TODO improve docs, this method is synchronous, until previous batch is processed another calls are being rejected (based on shouldRejectNextBatch behaviour). This imho contradicts the first line of this docs "AddEvents enqueues given events"
 func (client *DataSetClient) AddEvents(bundles []*add_events.EventBundle) error {
 	if client.finished.Load() {
 		return fmt.Errorf("client has finished - rejecting all new events")
 	}
-	errR := client.shouldRejectNextBatch()
-	if errR != nil {
+	if errR := client.shouldRejectNextBatch(); errR != nil { // TODO rename to client.lastBatchError(), should indicates return boolean type
+		// TODO document why we prevent of further processing in case of previous Error
+		// What happens in situation where
+		// 1. we enqueue reqA, reqB and reqC
```

Comment:
+1

Yeah, that's all the scenarios we discussed in the past, but the current behavior end to end (aka involving back-pressure in the otel collector and plugin rejection events) it's still not 100% clear to me.